### ISSUE TITLE
False positive for_kv_map in a consuming loop

### ISSUE THREAD

**Issue by Riateche** [rust-lang/rust-clippy#1279]
```
fn main() {
  let c: std::collections::HashMap<i32, String> = Default::default();
  for (_, b) in c {
    println!("{:?}", b);
  }
}
```

The lint suggests to use `values()` but AFAIK there is no consuming alternative to `values()`. (And if it exists, the message should mention it.)

Similar issue: #709.

**Comment by oli-obk** [comment]
Imo this is correct for copy types.

**Comment by Riateche** [comment]
> Imo this is correct for copy types.

indeed it may be. I've changed value type to `String` in the example, just to make it more clear.

**Comment by phansch** [comment]
If it is fixed, it would be nice to add this example to https://github.com/rust-lang/rust-clippy/blob/master/tests/ui/for_kv_map.rs as well.

**Comment by m-cat** [comment]
I'll raise a PR when I have a spare moment.
